pkg/network/usm/ebpf_javatls.go

```diff
@@ -36,195 +36,176 @@ const (
+		injectionBlockRegex: buildRegex(c.JavaAgentBlockRegex, "block"),
+		procRoot:            util.GetProcRoot(),
 	}
+
+	return res
 }
 
-func (p *JavaTLSProgram) Name() string {
+func (p *javaTLSProgram) Name() string {
 	return "java-tls"
 }
 
-func (p *JavaTLSProgram) IsBuildModeSupported(buildMode) bool {
+func (p *javaTLSProgram) IsBuildModeSupported(buildMode) bool {
 	return true
 }
 
-func (p *JavaTLSProgram) ConfigureManager(m *nettelemetry.Manager) {
-	p.manager = m
-	p.manager.Maps = append(p.manager.Maps, []*manager.Map{
+func (p *javaTLSProgram) ConfigureManager(m *nettelemetry.Manager) {
+	m.Maps = append(m.Maps, []*manager.Map{
 		{Name: javaTLSConnectionsMap},
 	}...)
 
-	p.manager.Probes = append(m.Probes,
+	m.Probes = append(m.Probes,
 		&manager.Probe{ProbeIdentificationPair: manager.ProbeIdentificationPair{
-			EBPFFuncName: "kprobe__do_vfs_ioctl",
+			EBPFFuncName: doVfsIoctlKprobeName,
 			UID:          probeUID,
 		},
 			KProbeMaxActive: maxActive,
 		},
 	)
-	rand.Seed(int64(os.Getpid()) + time.Now().UnixMicro())
-	authID = rand.Int63()
 }
 
-func (p *JavaTLSProgram) ConfigureOptions(options *manager.Options) {
+func (p *javaTLSProgram) ConfigureOptions(options *manager.Options) {
 	options.MapSpecEditors[javaTLSConnectionsMap] = manager.MapSpecEditor{
 		Type:       ebpf.Hash,
 		MaxEntries: p.cfg.MaxTrackedConnections,
 		EditorFlag: manager.EditMaxEntries,
 	}
 	options.MapSpecEditors[javaDomainsToConnectionsMap] = manager.MapSpecEditor{
 		Type:       ebpf.Hash,
-		MaxEntries: uint32(p.cfg.MaxTrackedConnections),
+		MaxEntries: p.cfg.MaxTrackedConnections,
 		EditorFlag: manager.EditMaxEntries,
 	}
 
 	options.ActivatedProbes = append(options.ActivatedProbes,
 		&manager.ProbeSelector{
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe__do_vfs_ioctl",
+				EBPFFuncName: doVfsIoctlKprobeName,
 				UID:          probeUID,
 			},
 		})
 }
 
-func (p *JavaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
+func (p *javaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
 	return []manager.ProbeIdentificationPair{
-		{EBPFFuncName: "kprobe__do_vfs_ioctl"},
-		{EBPFFuncName: "kprobe_handle_sync_payload"},
-		{EBPFFuncName: "kprobe_handle_close_connection"},
-		{EBPFFuncName: "kprobe_handle_connection_by_peer"},
-		{EBPFFuncName: "kprobe_handle_async_payload"},
+		{EBPFFuncName: doVfsIoctlKprobeName},
+		{EBPFFuncName: handleSyncPayloadKprobeName},
+		{EBPFFuncName: handleCloseConnectionKprobeName},
+		{EBPFFuncName: handleConnectionByPeerKprobeName},
+		{EBPFFuncName: handleAsyncPayloadKprobeName},
 	}
 }
 
 // isJavaProcess checks if the given PID comm's name is java.
 // The method is much faster and efficient that using process.NewProcess(pid).Name().
-func isJavaProcess(pid int) bool {
-	filePath := filepath.Join(util.GetProcRoot(), strconv.Itoa(pid), "comm")
+func (p *javaTLSProgram) isJavaProcess(pid int) bool {
```

Comment:
why do you need to pass by pointer?

Reply:
I need this function to be part of the program to access the members of the struct
the pointer receiver is not mandatory here, but a common practice in go say - if you have a single pointer receiver, for consistency, use pointer receiver everywhere